<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Brendan Law" />

<meta name="date" content="2019-03-06" />

<title>Intro to genotype wavelets</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Brendan's Masters Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/blaw36/Masters_Project">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Intro to genotype wavelets</h1>
<h4 class="author"><em>Brendan Law</em></h4>
<h4 class="date"><em>03/06/2019</em></h4>

</div>


<div id="load-snp-data" class="section level3">
<h3>Load SNP data</h3>
<p>24 SNPs, for 70 individuals. SNPs are single nucleotide polymorphism - a common type of genetic variation among people (must be seen in at least some x% threshold of the population to be a valid SNP). An SNP may be replacing a particular nucleotide of a DNA (C for a T, for example). See <a href="https://ghr.nlm.nih.gov/primer/genomicresearch/snp">here</a> for some details. Our data is genotype data at genetic variants (mainly SNPs), coded as a number, <span class="math inline">\(g \in \{0,1,2\}\)</span>, representing the number of copies of the minor allele (which may be one of the A,T,G,C I think?). An example illustrated in class showed how we may have, for two nucleotides, say AG, the possible combinations: AA, AG, GA, GG, with one of these being coded as a ‘minor allele’, and hence giving us either 0, 1 or 2 copies of it. Some (standard) imputation is then performed to generate probabilities of each combination, and the final number (a value between 0 and 2) represents the probability-weighted average of the number of copies of the minor allele at the single SNP of interest. <a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></p>
<pre><code>## $chr17.10159002
## 
##     0 0.020 0.021 0.025 0.031 0.034 0.037 0.042 0.052 0.053 0.058 0.068 
##    43     1     1     1     1     2     1     1     1     2     1     1 
## 0.069 0.082 0.087 0.089 0.122 0.129 0.233 0.531 0.543     1 
##     1     1     1     1     1     1     1     1     1     5 
## 
## $chr17.10159236
## 
##     0 0.002 0.003 0.368 0.378 0.379 0.502 0.631 0.702 0.776 0.785 0.791 
##    35     4     7     1     1     1     1     1     1     1     1     1 
## 0.834     1 1.626     2 
##     1    12     1     1 
## 
## $chr17.10160091
## 
##     0 0.000 0.195     1 1.000     2 
##    37    19     1    10     2     1 
## 
## $chr17.10160195
## 
##     0 0.002 0.003 0.004 0.345 0.346 0.518 0.740 0.997 0.999     1 1.003 
##    19     1     4     4     4     1     1     1     1     2    25     1 
## 1.440 1.835     2 
##     1     1     4 
## 
## $chr17.10160499
## 
##     0 0.002 0.003 0.022 0.026 0.078 0.082 0.090 0.091 0.097 0.098 0.120 
##    42     4     1     1     1     1     2     1     1     1     1     1 
## 0.144 0.154 0.160 0.165 0.176 0.222     1 
##     1     1     2     1     1     1     6 
## 
## $chr17.10160773
## 
##     0 0.000 0.462     1 1.000 
##    34    16     1    14     5 
## 
## $chr17.10161074
## 
##     0 0.000 0.461     1 1.000 
##    34    16     1    14     5 
## 
## $chr17.10161112
## 
##     0 0.000 0.339     1 1.000     2 
##    26    12     1    16     9     6 
## 
## $chr17.10161218
## 
##     0 0.014 0.017 0.021 0.025 0.027 0.030 0.031 0.042 0.045 0.053 0.057 
##    43     1     1     1     1     2     1     1     1     1     1     1 
## 0.060 0.072 0.076 0.090 0.098 0.122 0.219 0.375 0.468 0.473     1 
##     1     1     1     1     1     1     1     1     1     1     5 
## 
## $chr17.10161395
## 
##     0 0.002 0.003 0.004 0.460 0.994 0.997 0.998     1 1.000 
##    34     5    10     1     1     1     1     1    14     2 
## 
## $chr17.10161485
## 
##     0 0.002 0.003 0.004 0.055 0.076 0.090 0.322 0.935 0.972 0.987 0.988 
##    21     1     1     1     1     1     1     1     1     1     3     2 
## 0.989 0.991 0.995     1 1.082 1.357 1.974 1.975 1.977     2 
##     1     1     1    18     1     1     1     1     1     9 
## 
## $chr17.10162386
## 
##     0 0.002 0.004 0.040 0.054 0.076 0.090 0.342 0.956 0.982 0.991 0.992 
##    21     1     1     1     1     1     1     1     1     1     3     2 
## 0.993 0.994 0.999     1 1.086 1.574 1.982 1.983     2 
##     1     1     1    18     1     1     2     1     9 
## 
## $chr17.10162576
## 
##     0 0.002 0.003 0.024 0.033 0.042 0.067 0.713 0.970 0.975 0.976 0.979 
##    15     1     4     1     1     1     1     1     1     1     1     1 
## 0.984 0.991     1 1.000 1.003 1.093 1.796 1.976     2 
##     1     1    26     2     1     1     1     1     7 
## 
## $chr17.10162681
## 
##     0 0.000 0.379     1 1.000 
##    36    16     1    12     5 
## 
## $chr17.10162695
## 
##     0 0.002 0.378     1 1.000 
##    36    16     1    12     5 
## 
## $chr17.10162786
## 
##     0 0.000 0.302     1 1.000     2 2.000 
##    19     6     1    20    11     9     4 
## 
## $chr17.10162874
## 
##     0 0.000 0.376     1 1.000 
##    36    16     1    12     5 
## 
## $chr17.10162892
## 
##     0 0.002 0.003 0.004 0.027 0.029 0.150 0.180 0.200 0.245 0.326 0.336 
##    42     5     4     1     1     1     1     1     1     1     1     1 
## 0.425 0.438 0.457 0.808     1 
##     1     1     1     1     6 
## 
## $chr17.10163408
## 
##     0 0.003 0.338 0.797     1 1.001 1.004 1.006 1.041 
##    36    16     1     1    12     1     1     1     1 
## 
## $chr17.10163424
## 
##     0 0.002 0.003 0.310 0.786     1 1.000 1.001 1.034 
##    37     2    14     1     1    11     1     2     1 
## 
## $chr17.10163443
## 
##     0 0.002 0.003 0.304 0.782     1 1.000 1.001 1.033 
##    37     1    15     1     1    11     1     2     1 
## 
## $chr17.10163532
## 
##     0 0.062 0.063 0.111 0.120 0.132 0.157 0.158 0.568 0.783 0.987 0.991 
##    11     2     1     1     1     1     1     1     1     1     1     1 
## 0.992     1 1.012 1.036 1.038 1.049 1.076 1.083 1.515 1.896 1.897     2 
##     1    28     1     1     1     1     1     1     1     1     1     9 
## 
## $chr17.10163747
## 
##     0 0.000 0.002 0.209     1 1.000 
##    37    15     1     1    11     5 
## 
## $chr17.10163945
## 
##     0 0.000 0.361     1 1.000     2 
##    33    17     1    13     4     2</code></pre>
</div>
<div id="read-in-phenotype-data" class="section level3">
<h3>Read in phenotype data</h3>
<p>Counts at 1024 bases, for 70 individuals.</p>
<pre><code>## No id variables; using all as measure variables</code></pre>
<p><img src="Intro_genotype_wlets_files/figure-html/pressure-1.png" width="672" /><img src="Intro_genotype_wlets_files/figure-html/pressure-2.png" width="672" /><img src="Intro_genotype_wlets_files/figure-html/pressure-3.png" width="672" /></p>
</div>
<div id="combining-both" class="section level3">
<h3>Combining both</h3>
<p>The goal of this project is to: - Identify base locations which contain significant effects between individual SNPs and the count data obtained from sequencing - Estimate the effect of the SNP value on the sequenced count data at each base</p>
<p>So, this exercise involves taking two SNPs, for example, and trying to plot 70 individuals’ worth of counts, and giving different colours depending on their SNP values. A fairly fruitless task given how noisy the data is.</p>
<pre><code>## No id variables; using all as measure variables</code></pre>
<p><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-2-1.png" width="672" /><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
<pre><code>## No id variables; using all as measure variables</code></pre>
<p><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-2-3.png" width="672" /></p>
<p>Another way we can represent this is to, for example, bin our observations into the SNP they are closest to (based on the SNP data we’ve been given), and give an average count ‘function’. In a way, we’re reproducing a kind of figure seen on page 14 of the Shim and Stephens (2014) paper. That probably uses slightly adjusted count data (normalised for total read length at each base, covariates regressed out, etc, which we won’t do here). Take the significant, chr17.10161485, for example</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>## -- Attaching packages -------------------------------------------------------------------------------------------------------------------------------------------------- tidyverse 1.2.1 --</code></pre>
<pre><code>## v tibble  2.0.1     v purrr   0.3.0
## v tidyr   0.8.2     v dplyr   0.7.8
## v readr   1.3.1     v stringr 1.3.1
## v tibble  2.0.1     v forcats 0.3.0</code></pre>
<pre><code>## -- Conflicts ----------------------------------------------------------------------------------------------------------------------------------------------------- tidyverse_conflicts() --
## x dplyr::between()   masks data.table::between()
## x dplyr::filter()    masks plotly::filter(), stats::filter()
## x dplyr::first()     masks data.table::first()
## x dplyr::lag()       masks stats::lag()
## x dplyr::last()      masks data.table::last()
## x dplyr::select()    masks MASS::select(), plotly::select()
## x purrr::transpose() masks data.table::transpose()</code></pre>
<pre class="r"><code># Bin the individuals up
# SNP 2:
# chr17.10161485
pheno.and.gen.bin = cbind(pheno.dat,snp = as.numeric(as.character(eg_geno.2$chr17.10161485)))
pheno.and.gen.bin = data.frame(pheno.and.gen.bin)
pheno.and.gen.bin$bin = round(pheno.and.gen.bin$snp)
pheno.and.gen.bin.averaged = pheno.and.gen.bin %&gt;% 
  group_by(bin) %&gt;%
  summarise_all(funs(mean)) %&gt;%
  select(-&quot;snp&quot;)
pheno.and.gen.bin.averaged = melt(pheno.and.gen.bin.averaged, id.vars = &#39;bin&#39;)
pheno.and.gen.bin.averaged$loc = rep(1:1024, each = 3)

ggplot(pheno.and.gen.bin.averaged) + 
  geom_path(aes(x = loc, y = value, group = factor(bin), colour = factor(bin))) +
  labs(colour = &quot;Imputed SNP&quot;)</code></pre>
<p><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>A short guide to wavelet analysis on the sequenced count data. Let’s try and look at two individuals whose counts data are very different: one displays a lot of variation, the other does not: (note this is the raw wavelet count, NOT pre-processed to filter out low count WCs, or normalised against read counts, just to give a flavour of what wavelets on this data looks like).</p>
<pre><code>##  [1] 21 16 24 32 30 50 28 61 63 12 40  7 54 11 34 49 70  6 62  1 19 48 52
## [24]  4 37 29  5 43 39 17 35 64 58  9 41 31 20 10 59 13 51 46 66 25  2 14
## [47]  3 18 56 22 57 67 42 38 65 36 47 44 26 68 45 27 60 15 33 55 53 23  8
## [70] 69</code></pre>
<p>Individuals 21 (little variation) and 69 (a lot of variation)</p>
<pre><code>## No id variables; using all as measure variables</code></pre>
<p><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-5-1.png" width="672" /><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-5-2.png" width="672" /><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-5-3.png" width="672" /></p>
<pre><code>##  [1] 0.7071068 0.5000000 0.3535534 0.2500000 0.2651650 0.1875000 0.1325825
##  [8] 0.0625000 0.1104854 0.0468750</code></pre>
<p><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-5-4.png" width="672" /><img src="Intro_genotype_wlets_files/figure-html/unnamed-chunk-5-5.png" width="672" /></p>
<pre><code>##  [1] 2.121320 1.500000 1.944544 1.250000 1.856155 0.687500 1.679379
##  [8] 1.843750 1.458408 0.593750</code></pre>
<p>A more relevant exercise would be to take the difference of these two individuals, and then transform that into a wavelet space. Our exercise here is an attempt to model differences in the wavelet space, and then transform the differences back into the data space (in the form of beta coefficients - data space differences), so that the wavelet space (combined with certain choices of priors) can denoise some of the differences between the noisy signals. Let’s see what that looks like:</p>
<pre><code>## No id variables; using all as measure variables</code></pre>
<p><img src="Intro_genotype_wlets_files/figure-html/differences-1.png" width="672" /><img src="Intro_genotype_wlets_files/figure-html/differences-2.png" width="672" /></p>
<pre><code>##  [1] 2.121320 1.500000 1.944544 1.250000 1.944544 0.687500 1.635184
##  [8] 1.812500 1.347922 0.546875</code></pre>
<!-- References -->
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>See the Shim and Stephens (2014) paper for some more detail of the input data.<a href="#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
