<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Wavelet-based Genetic Association Analysis… - Shim and Stephens (2014)</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Masters Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/blaw36/Masters_Project">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Wavelet-based Genetic Association Analysis… - Shim and Stephens (2014)</h1>
<h4 class="date"><em>8 August 2018</em></h4>

</div>


<p>Full title: <em>Wavelet-based genetic association analysis of functional phenotypes arising from high-throughput sequencing assays</em></p>
<p>Paper can be found <a href="https://projecteuclid.org/euclid.aoas/1437397106">here</a></p>
<hr />
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<ul>
<li>High-level aim: Associate observable traits/behaviours with genetic variants - genetic association studies</li>
<li>Data is obtained from high-throughput sequencing assays, resulting in high-resolution data, which is not utilised by traditional association analysis methods (coarser resolution);
<ul>
<li>Fixed length windows</li>
<li>Genes</li>
</ul></li>
<li>Treat sequence data as measuring an underlying ‘function’ which varies on the genome
<ul>
<li>Use the high-res, wavelet-based methods for functional data analysis</li>
</ul></li>
<li>Focus in this paper is on identifying <em>genetic variants associated with chromatin accessibility</em>, using <em>DNase-seq</em></li>
<li>Wavelet-based test for <strong>association between a covariate</strong> <em>(say, a genotype)</em> <strong>and the shape of the underlying function</strong>
<ul>
<li>Also provide ability to estimate the shape of genotype efffect</li>
</ul></li>
<li>Data is wavelet transformed
<ul>
<li>Model associations in the transformed space</li>
<li>Spatial structure of data translates to sparse structure in wavelet space</li>
<li>Sparse data relatively easy to model</li>
<li><span style="color:red"><em>Is this related to the fact that all the localised ‘differences’ at finer granularities mainl show up as 0s due to the spatial structure of the data? Hence spatial similarity creates sparsity in the wavelet space, whereas signal is exposed as the ‘non-sparse’ area?</em></span></li>
</ul></li>
</ul>
</div>
<div id="background" class="section level3">
<h3>Background</h3>
<div id="dnase-seq-and-chromatin-accessibility" class="section level4">
<h4>DNase-seq and chromatin accessibility</h4>
<ul>
<li>Enzyme called DNase I selectively cuts DNA at locations where chromatin is accessible</li>
<li>More accessibility generally <span class="math inline">\(\Rightarrow\)</span> cut more often</li>
<li><span class="math inline">\(c_{b}\)</span> = count of reads starting at each base, <span class="math inline">\(b\)</span>, in the genome (humans: <span class="math inline">\(b \approx 1,\dots,3 \times 10^9\)</span>)</li>
<li><span class="math inline">\(d_{b} = \frac{c_{b}}{S}\)</span>, where <span class="math inline">\(S\)</span> is total number of mapped reads. This standardises the read counts</li>
<li>Generally, higher <span class="math inline">\(d_{b} \Rightarrow\)</span> higher accessibility of base <span class="math inline">\(b\)</span> (through DNase I sensitivity, which is a proxy)</li>
<li>Counts are 0 in many s=places, but where it exists, shows local spatial autocorrelation</li>
</ul>
</div>
<div id="wavelets" class="section level4">
<h4>Wavelets</h4>
<ul>
<li>Haar Discrete Wavelet Transform (<em>DWT</em>)
<ul>
<li>See <span style="color:red"><i>A theory for multiresolution signal decomposition&quot; by S.G. Mallat (1989)</i></span> for a more formal background</li>
</ul></li>
<li><span class="math inline">\(d = (d_b)^{B}_{b=1}\)</span> are standardised counts from a length <span class="math inline">\(B\)</span> region</li>
<li><span class="math inline">\(d\)</span> decomposed into “wavelet coefficients (WCs)”, <span class="math inline">\(y_{\textbf{s}cale,\textbf{l}ocation}\)</span>, and <span class="math inline">\(y=(y_{sl})\)</span>, the vector of all WCs.
<ul>
<li>Zero-cale: <span class="math inline">\(y_{01} = \sum_{b} d_{b}\)</span>. Sum of all</li>
<li>First scale: <span class="math inline">\(y_{11} = \sum_{b \leq \frac{B}{2}} d_{b} - \sum_{b &gt; \frac{B}{2}} d_{b}\)</span>, ie differences between the total counts of the ‘first’ and ‘second’ half of the sequence</li>
<li>This ‘method’ of differencing the ‘binary split’ regions generates finer data at ‘higher scale levels/numbers’</li>
</ul></li>
<li><span class="math inline">\(y = Wd\)</span>, and <span class="math inline">\(W\)</span> known as the <em>DWT matrix</em>.
<ul>
<li>One-one transform, hence <span class="math inline">\(W\)</span> is invertible, and we get the ‘inverse discrete wavelet transform’ (IDWT), <span class="math inline">\(d = W^{-1}y\)</span></li>
</ul></li>
<li>Two crucial properties:
<ol style="list-style-type: decimal">
<li>‘Whitening property’: WCs tend to be less dependent even if <span class="math inline">\(d\)</span> is strongly spatially correlated</li>
<li>Sparsity in ‘wavelet denoising’: Few big WCs, shrinking/ignoring smaller WCs can provide denoised estimates of signal</li>
</ol></li>
</ul>
</div>
</div>
<div id="methods" class="section level3">
<h3>Methods</h3>
<ul>
<li><span class="math inline">\(N\)</span> individuals</li>
<li>Local effect expected, hence DNase-seq data divided into regions, with each region tested for association with all near-by Single Nucleotide POlymorphisms (SNPs)</li>
<li><span class="math inline">\(d^i\)</span> is a vector of DNase-seq count data for individual <span class="math inline">\(i\)</span> (to <span class="math inline">\(N\)</span>). Has length <span class="math inline">\(B=2^J\)</span></li>
<li><span class="math inline">\(y^i\)</span> is a vector of WCs from DWT-transformed <span class="math inline">\(d\)</span>.</li>
<li><span class="math inline">\(g^i \in {0,1,2}\)</span> is <span class="math inline">\(\#\)</span> of copies of minor allele, at a single SNP of interest for inidividual <span class="math inline">\(i\)</span></li>
</ul>
<div id="test-setup" class="section level4">
<h4>Test setup</h4>
<ul>
<li><span class="math inline">\(H_0:\)</span> no association between any WC and <span class="math inline">\(g := \gamma_{sl} = 0 \forall s,l\)</span> (and thus, between <span class="math inline">\(d\)</span> and <span class="math inline">\(g\)</span> also)</li>
<li>Bayes Factor support for <span class="math inline">\(\gamma_{sl}=1\)</span>: (<span style="color:red"> read supplementary material for what is a Bayes Factor! But pretty much a statistic to conduct a hypothesis test in a Bayesian setting.</span>) <span class="math display">\[BF_{sl}(y,g)=\frac{p(y_{sl}|g,\gamma_{sl}=1)}{p(y_{sl}|g,\gamma_{sl}=0)}\]</span></li>
<li>Standard normal linear regression for <span class="math inline">\(p(y_{sl}|g,\gamma_{sl})\)</span>: <span class="math display">\[y_{sl}^i = \mu_{sl} + \gamma_{sl}\beta_{sl}g^i+\epsilon^i_{sl}\]</span> <span class="math display">\[\epsilon_{sl}^i \sim N(0,\sigma_{sl}^2)\]</span></li>
<li>See supplementary material for priors of mu, beta, sigma. Priors were chosen to be able to produce an analytic form solution.</li>
<li>Hierarchical model: <span class="math inline">\(p(\gamma_{sl}=1|\pi)=\pi_{s}\)</span>
<ul>
<li><span class="math inline">\(\pi_s\)</span> = proportion of WCs at <span class="math inline">\(s\)</span> associated with <span class="math inline">\(g\)</span></li>
<li><strong>Assumption of independence across scales and locations</strong></li>
<li>Likelihood ratio for <span class="math inline">\(\pi = (\pi_1,\dots,\pi_J)\)</span> relative to <span class="math inline">\(\pi \equiv 0\)</span> given on page 7. Full derivation involves joint over all <span class="math inline">\(s,l\)</span> and multiplying out the two states of <span class="math inline">\(\gamma_{sl}\)</span> <span class="math display">\[ \Lambda(\pi;y,g) = \prod_{s,l} [\pi_{s}BF_{sl}+(1-\pi_{s})]\]</span></li>
</ul></li>
<li>Null holds if <span class="math inline">\(\pi \equiv 0\)</span>, therefore, likelihood ratio statistic to test <span class="math inline">\(H_0\)</span> is: <span class="math display">\[\hat{\Lambda}(y,g) := \Lambda(\hat{\pi};y,g)\]</span></li>
<li><span class="math inline">\(\hat{\pi}\)</span> is the maximum likelihood estimate of <span class="math inline">\(\pi\)</span>, parameterised using an EM algo.</li>
<li>‘whitening property’: <span class="math inline">\(y_{sl}, \beta_{sl}\)</span> conditionally independent given <span class="math inline">\(\pi\)</span> across <span class="math inline">\(s,l\)</span> (moreso than the <span class="math inline">\(d\)</span>’s)</li>
</ul>
</div>
<div id="multiple-snps-permutation-procedure" class="section level4">
<h4>Multiple SNPs, permutation procedure</h4>
<ul>
<li>Permutation procedure to assess significance of observed <span class="math inline">\(\hat{\Lambda}\)</span></li>
<li>Single SNP with vector <span class="math inline">\(g\)</span> vs <span class="math inline">\(d\)</span>: <span class="math inline">\(\hat{\Lambda}\)</span> used to test this association</li>
<li>Association between <span class="math inline">\(d\)</span> and ANY of <span class="math inline">\(P\)</span> nearby SNPs, with <span class="math inline">\(g_1,\dots,g_P\)</span>: <span class="math display">\[ \hat{\Lambda}_{max} := \max_{p} \hat{\Lambda}(y,g_p) \]</span>
<ul>
<li>Assess significance by generating permutations <span class="math inline">\(\nu_1,\dots,\nu_M\)</span> of <span class="math inline">\((1,\dots,M)\)</span></li>
<li>Generating random combos of <span class="math inline">\(g^{i}\)</span>’s observed from each of the <span class="math inline">\(i\in(1,\dots,M)\)</span> individuals <span class="math display">\[ \hat{\Lambda}_{max}^j := \max_{p} \hat{\Lambda}(y,\nu_{j}(g_p)) \]</span> <span class="math display">\[ p-val_{\hat{\Lambda_{max}}} = \frac{\#\{j:\hat{\Lambda}_{max}^j \geq \hat{\Lambda}_{max}\}+1}{M+1} \]</span></li>
<li>This is effectively testing for significance by ‘simulating’ the statistic across a heap of other simulated/potential g_p vectors, and assessing the values of the resultant statistic</li>
</ul></li>
</ul>
</div>
<div id="filtering-low-count-wcs" class="section level4">
<h4>Filtering low count WCs</h4>
<ul>
<li>Just contribute noise (sampling error) and obscure the signal, so filter them out by allocating <span class="math inline">\(BF_{sl}=1\)</span>, implying no information about association (<span class="math inline">\(p(y|g)\)</span> same for both <span class="math inline">\(\gamma_{sl}=0,1\)</span>)</li>
<li>WC set <span class="math inline">\(\{y_{sl}^i\}_{i=1}^N\)</span> is “low count” if avg number of reads per individual is less than <span class="math inline">\(L=2\)</span> (for a given scale, location)
<ul>
<li>Tried 1,2,3,5,10. Performance identical for <span class="math inline">\(L \in \{2,3,5\}\)</span>, but not for 1,10, suggesting 1 was not filtering enough, 10 was filtering too much (perhaps).</li>
</ul></li>
</ul>
</div>
<div id="quantile-transformation-to-guard-against-non-normality" class="section level4">
<h4>Quantile transformation to guard against non-normality</h4>
<ul>
<li>Model assumes normally distributed residuals</li>
<li>Significant non-normal residuals can arise</li>
<li>To prevent these from distorting the signal, WCs are quantile-transformed to the quantiles of a standard normal distribution
<ul>
<li>This means they are ranked, and their ranking percentile (percent_rank) is converted to an equivalent value on a standard normal distribution (eg: 1, 10, 100 may become -3, 0, 3)</li>
<li>Guarantees that normality of residuals assumption holds under the null, and that Bayes Factors are well behaved under the null</li>
</ul></li>
</ul>
</div>
<div id="controlling-for-confounding-factors" class="section level4">
<h4>Controlling for confounding factors</h4>
<ul>
<li>PCA regression (regress the first principal components on the quantile transformed WCs), and take the residuals.</li>
<li>Residuals are then quantile transformed and used in the Bayes Factor calculations <span style="color:red">How? Are these used as <span class="math inline">\(y_i\)</span>’s?</span></li>
</ul>
</div>
<div id="effect-size-estimates" class="section level4">
<h4>Effect size estimates</h4>
<ul>
<li>All of this means we can derive posterior distributions on the effect sizes, but only based on data in the wavelet space.
<ul>
<li>Given <span class="math inline">\(\hat{\pi}\)</span>, we can derive <span class="math inline">\(p(\beta_{sl}|y,g,\hat{\pi})\)</span></li>
<li>Has a point mass at zero, and a 3-parameter t-distribution (see supplementary material)</li>
</ul></li>
<li>To make them meaningful, need to inverse discrete wavelet transform (IDWT) back to interpretable effects</li>
<li>Data form of the linear regression:
<ul>
<li>Each individual, <span class="math inline">\(i \in {1,\dots,n}\)</span> with wavelet coefficients from locations <span class="math inline">\(l\in{1,\dots,B}\)</span>, where <span class="math inline">\(B\)</span> is the length of the genome, allows us to write: <span class="math display">\[ Y = M + \beta g + E\]</span></li>
</ul></li>
<li><span class="math inline">\(Y\)</span> is a <span class="math inline">\(B \times N\)</span> matrix (each column is an individual, with a <span class="math inline">\(B \times 1\)</span> vector of observed counts along the genome, transformed into WCs)</li>
<li><span class="math inline">\(M\)</span> is a <span class="math inline">\(B \times N\)</span> matrix of means. Same with <span class="math inline">\(E\)</span> for error terms</li>
<li><span class="math inline">\(\beta\)</span> is <span class="math inline">\(B \times 1\)</span> - effect for each location</li>
<li><span class="math inline">\(g\)</span> is <span class="math inline">\(1 \times N\)</span> - differnt genotypes for each individual</li>
<li><span class="math inline">\(D = W^{-1}Y\)</span> from IDWT before. Pre-multiply by <span class="math inline">\(W^{-1}\)</span> to ‘convert’ our <span class="math inline">\(Y\)</span> into <span class="math inline">\(D\)</span>, and we get <span class="math display">\[ W^{-1}Y = W^{-1}(\dots) \\
\therefore D = W^{-1}M + W^{-1}\beta g + W^{-1}E \\
\therefore D = W^{-1}M + \alpha g + W^{-1}E \]</span></li>
<li>Because <span class="math inline">\(\alpha\)</span> has linear r/ship with <span class="math inline">\(\beta\)</span>, we get a closed form pointwise posterior mean and variance for <span class="math inline">\(\alpha_b\)</span>, where <span class="math inline">\(b = 1,\dots,B\)</span> (ie the effect size at each location on the genotype)</li>
</ul>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
