---
title: "effect sizes - all 24 SNPs"
author: "Brendan Law"
date: "20/08/2019"
header-includes:
   - \usepackage{bm}
   - \usepackage{amsmath}
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
input_data_path = "~/Cpp/WaveQTL_HMT/data/dsQTL/"
data_path <- "~/Cpp/WaveQTL_HMT/test/dsQTL/output/" # change this when you port it all over to the Masters Git repo
dataset <- "tree_tie_noQT"
waveqtl_data_path <- "~/Cpp/WaveQTL_HMT/test/dsQTL/output/WaveQTL/"
waveqtl_dataset <- "test.no.QT"
geno_select <- 11 # the one used in the demo

library(rmutil) # for beta-binomial distribution
library(dplyr)
# source("../code/sim3_functions.R")
source("code/sim3_functions.R")
```
## WaveQTL_HMT vs WaveQTL
The idea is that the benefits will come from being able to detect signals which are not necessarily strong, and not necessarily broad, but relatively localised. These sorts of signals won't be captured by WaveQTL as they may be 'split' or 'spread' across a particular scale, and not propagate strongly enough to lower scales. (???)

Perhaps see Shim and Stephens figures for some examples. Perhaps some of that matter around the middle of Figure 4 may be able to be better captured using a HMT prior. There is potentially some difference between groups here, but very narrow, inconsistent, and seemingly very spiky signal.

### Read in phenotype (sequencing count) data, and other auxillary data
Here is just a sample codebase. We're working off the data in the WaveQTL git repo -- DNase-seq data at chr17.10160989.10162012
and genotypes at 24 SNPs in 2kb cis-candidate region on 70 individuals.
```{r}
pheno.dat = as.matrix(read.table(paste0(input_data_path, "chr17.10160989.10162012.pheno.dat")))
dim(pheno.dat)
#[1]   70 1024

### Is this useful at all? For later on
## read library read depth
library.read.depth = scan(paste0(input_data_path, "library.read.depth.dat"))
length(library.read.depth)

## read Covariates
Covariates = as.matrix(read.table(paste0(input_data_path, "PC4.dat")))

## Read DWT matrix 
Wmat_1024 = read.table("~/Cpp/WaveQTL/data/DWT/Wmat_1024",as.is = TRUE)
W2mat_1024 = Wmat_1024*Wmat_1024

# ## Read in SNPs
# geno_data = read.table("~/Cpp/WaveQTL/data/dsQTL/chr17.10160989.10162012.2kb.cis.geno",as.is = TRUE)
# geno_data = geno_data[11,4:73]
#
# # Group based on midpoint of the data
# med_data <- median(as.numeric(as.vector(geno_data[1,])))
# group_data <- as.numeric(as.numeric(as.vector(geno_data[1,])) >= med_data)
# write.table(t(c("blah","A","A",group_data)), file= paste0("~/Cpp/WaveQTL_HMT/data/dsQTL/", "sim3.cis.geno"), row.names=FALSE, col.names = FALSE, quote=FALSE)

# Generate own SNPs data
group_data <- rbinom(n = 70,size = 1,prob = 0.5)
write.table(t(c("blah","A","A",group_data)), file= paste0("~/Cpp/WaveQTL_HMT/data/dsQTL/", "sim3.cis.geno"), row.names=FALSE, col.names = FALSE, quote=FALSE)
```

Now, summarise the counts, and average:
```{r}
# Count summation
seq_sum <- apply(pheno.dat,MARGIN = 2,sum)
# Count average
seq_avg <- apply(pheno.dat,MARGIN = 2,mean)
plot(seq_sum, type = "l")
plot(seq_avg, type = "l")
```

### Simulate effect sizes and effect locations.
Trying to simulate effect sizes for all the 24 genos
```{r}
waveqtl_effect_size_list <- list()
for(i in 1:24){
  waveqtl_effect_size_list[[i]] <- with_hmt_effect_size(data_path = data_path
                                           ,dataset = dataset
                                           ,waveqtl_dataset = paste0("WaveQTL/",waveqtl_dataset)
                                           ,Wmat_1024 = Wmat_1024
                                           ,geno_select = i
                                           ,plot_title = paste0("Posterior mean +/3 posterior standard deviaion - SNP ",i))
}
```

waveqtl_non_hmt
```{r}
non_hmt_effect <- function(
  sel_geno_IX = 11
  ,Wmat_1024 = Wmat_1024
  ,W2mat_1024 = W2mat_1024
  ,plot_title = "Posterior mean +/3 posterior standard deviaion - SNP "
){
  ## Read posterior mean and varaince of effect sizes in Wavelet space
  ## Set a path to files
  beta_mean_path= "~/Cpp/WaveQTL/test/dsQTL/output/test.no.QT.fph.mean.txt"
  beta_var_path = "~/Cpp/WaveQTL/test/dsQTL/output/test.no.QT.fph.var.txt"
  
  ## Read posterior mean in Wavelet space and transform them back to data space 
  beta_mean = as.numeric(read.table(beta_mean_path)[sel_geno_IX,2:1025])
  beta_dataS = as.vector(-matrix(data=beta_mean, nr = 1, nc = 1024)%*%as.matrix(Wmat_1024))
  length(beta_dataS)
  ## [1] 1024
  beta_dataS[1:6]
  ## [1] 1.352119e-11 1.352119e-11 1.352119e-11 1.352119e-11 1.352119e-11
  ## [6] 1.352119e-11
  
  
  
  ## Read posterior variance in Wavelet space, transform them back to data space, and get standard deviation
  beta_var = as.numeric(read.table(beta_var_path)[sel_geno_IX,2:1025])
  beta_var_dataS = as.vector(matrix(data=beta_var, nr = 1, nc = 1024)%*%as.matrix(W2mat_1024))
  beta_sd_dataS = sqrt(beta_var_dataS)
  length(beta_sd_dataS)
  ## [1] 1024
  beta_sd_dataS[1:6]
  ## [1] 3.482833e-11 3.482833e-11 3.482833e-11 3.482833e-11 3.482833e-11
  ## [6] 3.482833e-11
  
  
  
  
  ## Visualize estimated effect size in the data space
  ymin_beta = min(beta_dataS - 3*beta_sd_dataS) - abs(min(beta_dataS - 3*beta_sd_dataS))*0.0000000001
  ymax_beta = max(beta_dataS + 3*beta_sd_dataS) + abs(max(beta_dataS + 3*beta_sd_dataS))*0.0000000001
  
  beta_l = beta_dataS - 3*beta_sd_dataS
  beta_r = beta_dataS + 3*beta_sd_dataS
  
  wh_l = which(beta_l > 0)
  wh_r = which(beta_r < 0)
  high_wh = sort(unique(union(wh_l, wh_r)))
  
  xval = 1:1024
  col_posi = xval[high_wh]
  
  # pdf("../test/dsQTL/effectSize.pdf", width = 8, height=3)
  par(mar = c(2,4,4,2))
  plot(1,1,type="n", xlab = "position", ylab = "Effect size",ylim=c(ymin_beta, ymax_beta),xlim=c(1, 1024),main =plot_title, axes=FALSE)
  axis(2)
  if(length(col_posi) > 0){
    for(j in 1:length(col_posi)){
      polygon(c(col_posi[j]-0.5, col_posi[j]-0.5, col_posi[j]+0.5, col_posi[j]+0.5), c(ymin_beta-2, ymax_beta+2, ymax_beta+2, ymin_beta-2), col ="pink", border = NA)
    }
  }
  
  abline(h = 0, col = "red")
  points(xval, beta_dataS, col = "blue", type="l")
  points(xval, beta_l, col = "skyblue", type="l")
  points(xval, beta_r, col = "skyblue", type="l")
  box()
  
  # dev.off()
  p <- recordPlot()
  return(p)
}



waveqtl_effect_size_list_nonhmt <- list()
for(i in 1:24){
  waveqtl_effect_size_list_nonhmt[[i]] <- non_hmt_effect(sel_geno_IX = i
                                                         ,Wmat_1024 = Wmat_1024
                                                         ,W2mat_1024 = W2mat_1024
                                                         ,plot_title = paste0("Posterior mean +/3 posterior standard deviaion - SNP ",i))
}
```

```{r}
waveqtl_effect_size_list[[17]]$p
waveqtl_effect_size_list_nonhmt[[17]]
```

```{r}
effect_larger_seq_sum <- readRDS("~/Cpp/WaveQTL_HMT/sim3_results/sim3_50_sum_seq_scale.RDS")
```
